main.tf

```diff
@@ -24,58 +26,150 @@ resource "tls_self_signed_cert" "linkerd-ca" {
   }
 }
 
-resource "tls_cert_request" "linkerd-cert-request" {
-  key_algorithm   = tls_private_key.linkerd["issuer"].algorithm
-  private_key_pem = tls_private_key.linkerd["issuer"].private_key_pem
+resource "tls_self_signed_cert" "webhook_anchor" {
+  key_algorithm     = tls_private_key.linkerd["webhook_anchor"].algorithm
+  private_key_pem   = tls_private_key.linkerd["webhook_anchor"].private_key_pem
+  is_ca_certificate = true
+
+  validity_period_hours = var.trust_anchor_validity_hours
+  allowed_uses          = ["cert_signing", "crl_signing"]
 
   subject {
-    common_name = "identity.linkerd.cluster.local"
+    common_name = "webhook.linkerd.cluster.local"
   }
 }
 
-resource "tls_locally_signed_cert" "linkerd-cert" {
-  is_ca_certificate = true
-
-  cert_request_pem = tls_cert_request.linkerd-cert-request.cert_request_pem
-
-  ca_key_algorithm   = tls_private_key.linkerd["trust_anchor"].algorithm
-  ca_private_key_pem = tls_private_key.linkerd["trust_anchor"].private_key_pem
-  ca_cert_pem        = tls_self_signed_cert.linkerd-ca.cert_pem
-
-  validity_period_hours = var.issuer_validity_hours
-  allowed_uses          = ["cert_signing", "crl_signing"]
+resource "kubernetes_namespace" "namespace" {
+  for_each = local.namespace
+  metadata {
+    name = each.key
+    annotations = {
+      "linkerd.io/inject" = "enabled"
+    }
+  }
 }
-#
-# END
-
-resource "helm_release" "linkerd" {
-  name             = "linkerd"
-  chart            = "${var.chart_repository}/linkerd2-${var.chart_version}.tgz"
-  version          = var.chart_version
-  create_namespace = true
 
-  set {
-    name  = "global.identityTrustAnchorsPEM"
-    value = tls_self_signed_cert.linkerd-ca.cert_pem
+resource "kubernetes_secret" "webhook_trust_anchor" {
```

Comment:
Why are we creating secrets in K8s with this information?  Is that required by the helm chart?

Reply:
need to create the secrets before installing the helm charts